components/clarinet-format/src/formatter/mod.rs

```diff
@@ -0,0 +1,1220 @@
+                                        format!(
+                                            " {}",
+                                            &self.display_pse(comment, previous_indentation)
+                                        )
+                                    } else if iter.peek().is_some() {
+                                        " ".to_string()
+                                    } else {
+                                        "".to_string()
+                                    }
+                                )
+                            }
+                        }
+                    } else if let Some(define) = DefineFunctions::lookup_by_name(atom_name) {
+                        match define {
+                            DefineFunctions::PublicFunction
+                            | DefineFunctions::ReadOnlyFunction
+                            | DefineFunctions::PrivateFunction => self.format_function(list),
+                            DefineFunctions::Constant
+                            | DefineFunctions::PersistedVariable
+                            | DefineFunctions::NonFungibleToken
+                            | DefineFunctions::FungibleToken => self.format_constant(list),
+                            DefineFunctions::Map => self.format_map(list, previous_indentation),
+                            DefineFunctions::UseTrait | DefineFunctions::ImplTrait => {
+                                // these are the same as the following but need a trailing newline
+                                format!(
+                                    "({})\n",
+                                    self.format_source_exprs(list, previous_indentation)
+                                )
+                            }
+                            // DefineFunctions::Trait => format_trait(settings, list),
+                            // DefineFunctions::PersistedVariable
+                            // DefineFunctions::FungibleToken
+                            // DefineFunctions::NonFungibleToken
+                            _ => self.format_source_exprs(&[expr.clone()], previous_indentation),
+                        }
+                    } else {
+                        self.to_inner_content(list, previous_indentation)
+                    };
+                    result.push_str(t(&formatted));
+                    continue;
+                }
+            }
+            let current = self.display_pse(expr, previous_indentation);
+            let mut between = " ";
+            if let Some(next) = iter.peek() {
+                if !is_same_line(expr, next) || is_comment(expr) {
+                    between = "\n";
+                }
+            } else {
+                // no next expression to space out
+                between = "";
+            }
+
+            result.push_str(&format!("{current}{between}"));
+        }
+        result
+    }
+
+    fn format_constant(&self, exprs: &[PreSymbolicExpression]) -> String {
```

Comment:
I tried
```rs
let result = format_with_default(&String::from("(define-fungible-token hello)"));
```

But it seems to fail.
The second param is options in `define-fungible-token`

Reply:
Didn't realize that. easy fix. It uses this dumb function called `name_and_args` which i should really remove because it's just asking for trouble